lib/stellar/sep10.rb

```diff
@@ -0,0 +1,353 @@
+        )
+      end
+
+      if transaction.operations.size != 1
+        raise InvalidSep10ChallengeError.new(
+          "The transaction should contain only one operation"
+        )
+      end
+
+      operation = transaction.operations.first
+      client_account_id = operation.source_account
+
+      if client_account_id.nil?
+        raise InvalidSep10ChallengeError.new(
+          "The transaction's operation should contain a source account"
+        )
+      end
+
+      if operation.body.arm != :manage_data_op
+        raise InvalidSep10ChallengeError.new(
+          "The transaction's operation should be manageData"
+        )
+      end
+
+      if operation.body.value.data_value.unpack("m")[0].size !=  48
+        raise InvalidSep10ChallengeError.new(
+          "The transaction's operation value should be a 64 bytes base64 random string"
+        )
+      end
+
+      if !verify_tx_signed_by(transaction_envelope: envelope, keypair: server)
+        raise InvalidSep10ChallengeError.new(
+          "The transaction is not signed by the server"
+        )
+      end
+
+      time_bounds = transaction.time_bounds
+      now = Time.now.to_i
+
+      if time_bounds.nil? || !now.between?(time_bounds.min_time, time_bounds.max_time)
+        raise InvalidSep10ChallengeError.new("The transaction has expired")        
+      end
+
+      # Mirror the return type of the other SDK's and return a string
+      client_kp = Stellar::KeyPair.from_public_key(client_account_id.ed25519!)
+
+      return envelope, client_kp.address
+    end
+
+
+    Contract(C::KeywordArgs[
+      challenge: String,
+      server: Stellar::KeyPair,
+      signers: ArrayOf[Stellar::AccountSigner]
+    ] => C::ArrayOf[Stellar::AccountSigner])
+    # Verifies that for a SEP 10 challenge transaction all signatures on the transaction are accounted for.
+    #
+    # A transaction is verified if it is signed by the server account, and all other signatures match a signer 
+    # that has been provided as an argument. Additional signers can be provided that do not have a signature, 
+    # but all signatures must be matched to a signer for verification to succeed. 
+    #
+    # If verification succeeds a list of signers that were found is returned, excluding the server account ID.
+    #
+    # @param challenge [String] SEP0010 transaction challenge transaction in base64.
+    # @param server [Stellar::Keypair] keypair for server's account.
+    # @param signers [ArrayOf[Stellar::AccountSigner]] The signers of client account.
+    #
+    # @return [ArrayOf[Stellar::AccountSigner]]
+    #
+    # Raises a InvalidSep10ChallengeError if:
+    #     - The transaction is invalid according to Stellar::SEP10.read_challenge_transaction.
+    #     - One or more signatures in the transaction are not identifiable as the server account or one of the 
+    #       signers provided in the arguments.
+    def self.verify_challenge_transaction_signers(
```

Comment:
Can we name this `verify_challenge_tx_signers` with `tx` instead of `transaction` to match the existing `build_challenge_tx` function?